server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/vacation/VacationResponseSet.scala

```diff
@@ -0,0 +1,62 @@
+package org.apache.james.jmap.vacation
+
+import java.time.ZonedDateTime
+
+import org.apache.james.jmap.api.vacation.VacationPatch
+import org.apache.james.jmap.model.AccountId
+import org.apache.james.jmap.model.State.State
+import play.api.libs.json.JsObject
+
+case class VacationResponseSetRequest(accountId: AccountId,
+                               update: Map[String, VacationResponsePatchObject]) {
+  def parsePatch(): Either[IllegalArgumentException, VacationResponsePatchObject] = {
+    update.map({
+      case (id, patch) => id match {
+        case "singleton" => Right(patch)
+        case _ => Left(new IllegalArgumentException("id must be singleton"))
+      }
+    }).head
+  }
+}
+
+case class VacationResponseId(value: String)
```

Comment:
Can we enforce via the typing that this is always "singletion" ? (Via a value object?)

Reply:
Or through the serializer... (which is what I did for the get)

Reply:
if the map is empty you will throw a runtime exception not handled by the Either when calling `.head`
Otherwise having a `case class VacationResponseSetRequest(accountId: AccountId,
                               patch: VacationResponsePatchObject)`
and handling the convertion in the serializer is good for me.